src/LookupTable/containers/LookupTablePage.tsx

```diff
@@ -24,45 +21,39 @@ const LookupTablePage: React.FC<{ basePath: string }> = ({ basePath = '' }) => {
 
   const { lookupTableID: structureID } = params
 
-  const { data, loading, error, refetch } = useGetLookupTableStructureByIdQuery({
-    variables: { lookupTableID: Number(structureID) },
-    fetchPolicy: 'network-only',
-  })
-
-  useEffect(() => {
-    if (data?.dataTable) {
-      const lookupTable = data.dataTable as LookUpTableType
-      setStructure(lookupTable)
-    }
-  }, [data])
+  const {
+    tableStructure,
+    tableStructureLoadingState: { error, loading },
+    refetchTableStructure,
+  } = useGetSingleTableStructure(structureID)
 
-  const dataViewCode = structure?.dataViewCode
+  const { dataViewCode, displayName, id } = tableStructure ?? {}
 
   return (
     <React.Fragment>
       {error ? (
-        <Message error header={t('LOOKUP_ERROR_TITLE')} list={[error.message]} />
+        <Message error header={t('LOOKUP_ERROR_TITLE')} list={[error]} />
       ) : loading ? (
         <Loading />
-      ) : structure ? (
+      ) : tableStructure ? (
         <Container style={{ padding: '2em 0em' }}>
-          <LookUpMainMenu tableLabel={structure?.displayName} tableId={structure?.id} />
+          <LookUpMainMenu tableLabel={displayName} tableId={id} />
           <Divider />
-          {/* Show data view table if defined, otherwise default table display */}
           {dataViewCode && <DataViewTable codeFromLookupTable={dataViewCode} />}
-          {!structure.dataViewCode && <LookUpTable structure={structure} />}
+          {!dataViewCode && <p>Lookup table has no dataViewCode defined</p>}
```

Comment:
Didn't bother localising this because it should never actually appear if everything else is working correctly.